src/lib_json/json_value.cpp

```diff
@@ -1168,13 +1168,35 @@ Value const& Value::operator[](CppTL::ConstString const& key) const {
   return *found;
 }
 #endif
-
 Value& Value::append(const Value& value) { return (*this)[size()] = value; }
 
 Value& Value::append(Value&& value) {
   return (*this)[size()] = std::move(value);
 }
 
+/// \brief Insert value in array at specific index
+bool Value::insert(ArrayIndex index, const Value& newValue) {
+  JSON_ASSERT_MESSAGE(type() == nullValue || type() == arrayValue,
+                      "in Json::Value::insert(ArrayIndex index, const Value& "
+                      "newValue): requires arrayValue");
+  return insert(index, Value(newValue));
+}
+bool Value::insert(ArrayIndex index, Value&& newValue) {
+  JSON_ASSERT_MESSAGE(type() == nullValue || type() == arrayValue,
+                      "in Json::Value::insert(ArrayIndex index, Value&& "
+                      "newValue): requires arrayValue");
+  ArrayIndex length = size();
+  if (index > length) {
```

Comment:
What if size is 10, and index is 20? Do we want to add nine blank entries?

Reply:
I wish it will be appended to the end of array to avoid blank entries.

Reply:
It is a question.
If the index is bigger than size, is it better to throw a exception or error?
I am not sure so that I invoke the append method. Maybe it should be removed and throw a exception.

Reply:
> What if size is 10, and index is 20? Do we want to add nine blank entries?
@baylesj 
in this case, should we throw an exception or just return false?
Appending to the tail may be not good.

Reply:
change to return false only, like the method `removeMember`
```
  if (type() != objectValue) {
    return false;
  }
```